src/org/traccar/database/LdapProvider.java

```diff
@@ -16,36 +16,161 @@
 package org.traccar.database;
 
 import javax.naming.Context;
+import javax.naming.NamingEnumeration;
 import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
 import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import org.traccar.Config;
+import org.traccar.helper.Log;
+import org.traccar.model.User;
+
 import java.util.Hashtable;
 
 public class LdapProvider {
 
     private String url;
-    private String context;
+    private String searchBase;
+    private String idAttribute;
+    private String nameAttribute;
+    private String mailAttribute;
+    private String searchFilter;
+    private String adminFilter;
+    private String serviceUser;
+    private String servicePassword;
 
-    public LdapProvider(String url, String context) {
-        this.url = url;
-        this.context = context;
+    public LdapProvider(Config config) {
+        String url = config.getString("ldap.url");
+        if (url != null) {
+            this.url = url;
+        } else {
+            this.url = "ldap://" + config.getString("ldap.server") + ":" + config.getInteger("ldap.port", 389);
+        }
+        this.searchBase = config.getString("ldap.base");
```

Comment:
Search base is the way to limit search scope, to not lookup full domain, but some OU only for example
`ou=Users,ou=Branch,dc=domain,dc=local`